src/cluster.rs

```diff
@@ -203,25 +252,25 @@ impl Clustering {
      * Retrieve the stats achieved from the clustering process.
      * Returns as many values as the number of iterations made.
      */
-    pub fn iteration_stats(&self) -> Result<&[FaissClusteringIterationStats]> {
+    pub fn iteration_stats(&self) -> &[ClusteringIterationStats] {
         unsafe {
             let mut data = ptr::null_mut();
             let mut size = 0;
             faiss_Clustering_iteration_stats(self.inner, &mut data, &mut size);
-            Ok(::std::slice::from_raw_parts(data, size))
+            ::std::slice::from_raw_parts(data as *mut ClusteringIterationStats, size)
```

Comment:
data is `*mut Faiss ClusteringIterationStats`, cast to `*mut ClusteringIterationStats`.

`ClusteringIterationStats` is an opaque type which is manually created.